vms/avm/txs/mempool/mempool.go

```diff
@@ -62,60 +65,58 @@ type Mempool interface {
 }
 
 type mempool struct {
-	bytesAvailableMetric prometheus.Gauge
-	bytesAvailable       int
-
-	unissuedTxs linkedhashmap.LinkedHashmap[ids.ID, *txs.Tx]
-	numTxs      prometheus.Gauge
-
-	toEngine chan<- common.Message
+	lock           sync.RWMutex
+	unissuedTxs    linkedhashmap.LinkedHashmap[ids.ID, *txs.Tx]
+	consumedUTXOs  set.Set[ids.ID]
+	bytesAvailable int
```

Comment:
I restructured the order of the fields so that all the fields protected by the lock could be cleanly put under it.